stan/math/prim/mat/fun/resize.hpp

```diff
@@ -6,49 +6,32 @@
 
 namespace stan {
   namespace math {
-
     namespace {
-
-      template <typename T>
-      void resize(Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>& x,
-                  const std::vector<size_t>& dims,
-                  size_t pos) {
-        x.resize(dims[pos], dims[pos + 1]);
-      }
-
-      template <typename T>
-      void resize(Eigen::Matrix<T, Eigen::Dynamic, 1>& x,
-                  const std::vector<size_t>& dims,
-                  size_t pos) {
-        x.resize(dims[pos]);
-      }
-
-      template <typename T>
-      void resize(Eigen::Matrix<T, 1, Eigen::Dynamic>& x,
-                  const std::vector<size_t>& dims,
-                  size_t pos) {
-        x.resize(dims[pos]);
+      template <typename T, int R, int C>
+      void resize(Eigen::Matrix<T, R, C>& x,
```

Comment:
It didn't look like resize was being used, so I took the liberty of adopting it and modifying it to suite my needs.

Need 1 was changing it so that the resize semantics match the dim semantics. Previous, to resize a RowVector or a Vector to something like length N, you could just pass in {N}. Now to be consistent with dims, you need to pass in {1, N} for RowVector or {N, 1} for Vector (where {} are std::vectors)

Also I switched the size_ts to ints. This is also to match the dims function.